packages/core/src/odata-v2/request-builder/get-by-key-request-builder.spec.ts

```diff
@@ -119,6 +140,34 @@ describe('GetByKeyRequestBuilder', () => {
       expect(actual.data.d).toEqual(entityData);
       expect(actual.request.method).toBe('GET');
     });
+
+    it('returns navigation properties', async () => {
```

Comment:
I think this test is unnecessary. In the other test we are checking that the url is correct. In this one we are mocking data for this url and checking that this data comes back when invoking the url. As we also have an E2E test I don't see a good reason for this test.
Is there something tested that I don't see?

Reply:
As you said, the  most unit tests in this file test the `url`/`execute` methods for verifying the final URL of the request.
As `appendPath` can only use `executeRaw`, this test is the replacement of the one for `execute`.

Reply:
I would then make this test as simple as possible and only check the URL from `url`/`executeRaw`.

Reply:
But what is this test checking in addition to the one above? I don't see any difference except, that we check, that `executeRaw` is behaving the way we expect (which should already be tested in other tests as this is not new functionality).
Maybe it would be worth testing `relativeUrl` instead.

Reply:
As discussed, this test will be kept for testing the `appendPath` API in the request builder. The tests for `releativeUrl` is added for lighter tests.